compiler/src/yul/names.rs

```diff
@@ -39,6 +39,12 @@ pub fn decode_name<T: AbiEncoding>(typ: &T, location: AbiDecodeLocation) -> yul:
     identifier! { (full_name) }
 }
 
+/// Generates an external call function name for a given type and location.
+pub fn contract_call(contract_name: String, func_name: String) -> yul::Identifier {
```

Comment:
It may be slightly more idiomatic (and efficient) to take these as `&str`. But our code base isn't consistent with this so far so I don't actually care that much. At some point when the project gets more mature we'll probably tackle these things. That said, in cases where taking `&str` doesn't complicate anything (e.g. doesn't depend on introducing life time syntax) I'd advocate for always preferring taking a `&str` not a `String`

Reply:
Agreed, too much value passing going on in the codebase.